syncer/sharding_group.go

```diff
@@ -119,40 +127,28 @@ func NewShardingGroup(sources []string, isSchemaOnly bool) *ShardingGroup {
 
 // Merge merges new sources to exists
 // used cases
-//   * add a new database / table to exists sharding group
+//   * add a new table to exists sharding group
 //   * add new table(s) to parent database's sharding group
-//  if group is un-resolved, we add it in sources and set it true
+//  if group is in sequence sharding, return error directly
 //  othereise add it in source, set it false and increment remain
 func (sg *ShardingGroup) Merge(sources []string) (bool, bool, int, error) {
 	sg.Lock()
 	defer sg.Unlock()
 
-	// need to check whether source is exist? but we maybe re-sync more times
-	isResolving := sg.remain != len(sg.sources)
-	ddls := []string{"create table"}
+	// NOTE: we don't support add shard table when in sequence sharding
+	if sg.meta.InSequenceSharding() {
+		return true, sg.remain <= 0, sg.remain, errors.NotSupportedf("in sequence sharding, can't add table")
+	}
 
 	for _, source := range sources {
-		synced, exist := sg.sources[source]
-		if isResolving && !synced {
-			if exist {
-				if !synced {
-					sg.remain--
-				} else if !utils.CompareShardingDDLs(sg.sourceDDLs[source], ddls) {
-					return isResolving, sg.remain <= 0, sg.remain, errors.NotSupportedf("execute multiple ddls: previous ddl %s and current ddls %q for source table %s", sg.sourceDDLs[source], ddls, source)
-				}
-			}
-
-			sg.sources[source] = true
-			sg.sourceDDLs[source] = ddls // fake create table ddl
-		} else {
-			if !exist {
-				sg.remain++
-			}
+		_, exist := sg.sources[source]
+		if !exist {
+			sg.remain++
 			sg.sources[source] = false
 		}
 	}
 
-	return isResolving, sg.remain <= 0, sg.remain, nil
+	return false, sg.remain <= 0, sg.remain, nil
```

Comment:
can we support replicating `CREATE TABLE` statement for a fresh group now?

Reply:
what is `fresh group` meant for

Reply:
do we support it before this pr

Reply:
the logic doesn't change, if `create table` with a new sharding group, just create a new one directly
https://github.com/pingcap/dm/blob/2ff99b5a7673a8325d748a81b54d4fd7bbed2c03/syncer/sharding_group.go#L419-L423

Reply:
the false (needShardingHandle) in this return will ignore the statement? then no table will be created in the downstream, right?

Reply:
not test, if so, the old code has this bug too

Reply:
it seems strange for no tables in one shard. For example
1. have two upstreams, each upstream has no shard tables, there exists no target table in downstream too. Then if one of upstream creates one shard table and applies sharding handling, send DDLInfo to DM-master. But DM-master has to wait for the other DM-master for this create table DDL. It seems we can't use common shard way for `create table`

Reply:
en, seems a bug 😢